### DISCUSSION TITLE
Mac M2,  How can I upgrade Torch in the "automatic" environment?

### DISCUSSION THREAD

**Discussion by sevenlucas7** [vladmandic/sdnext#996] [Q&A]
How can I update Torch to version 2.0.1 in the "automatic" environment on Mac M2? I already have the nightly version installed in my system, but the Torch version in the "automatic" environment is 2.0. I couldn't find any documentation that specifies how to set the Torch version for the A1111 program. Can you please guide me on how to update it?

**Comment by vladmandic** [comment]
if you're on one of latest version, running `webui --reinstall` should do the trick as it will reinstall latest compatible version of all packages.

**Reply by brknsoul** [reply]
will `--reinstall` overwrite any settings (ui-config.json, styles,csv, etc)?

**Reply by vladmandic** [reply]
no, all user settings remain. but i just took a look at the code and torch 2.0.1 was unlocked for cuda systems only, not for m1/m2. give me an hour, i'll unlock it.

**Reply by brknsoul** [reply]
Is torch 2.0.1 out for amd?

**Reply by vladmandic** [reply]
not yet.

**Reply by fanshu7** [reply]
Thank you for your answer. I'll give it a try

**Comment by sevenlucas7** [comment]
After running "./webui.sh --reinstall" and opening the web page, I still see Torch as “2.0.0 with Autocast half" enabled. Do I need to perform any additional steps? Looking forward to your response. Thank you.

**Reply by sevenlucas7** [reply]
<img width="754" alt="iShot_2023-05-17_18 16 41" src="https://github.com/vladmandic/automatic/assets/131904649/9632270b-22e0-4c85-952a-7aa33d31e139">

**Reply by vladmandic** [reply]
i just locked it down to 2.0.0 on mac as there are reports that 2.0.1 is causing problems on m1.

**Reply by sevenlucas7** [reply]
But when I run Torch 2.1 nightly on A1111's web UI, no issues occur.

**Reply by vladmandic** [reply]
nightly might as well be ok. 2.0.1 doesn't. 
you're totally free to install any version you want and i appreciate the feedback. but i cannot recommend installing nightly in installer itself :)

**Reply by sevenlucas7** [reply]
Thank you for your suggestion. I just want to improve the efficiency of M2. Now, with the added plugins like Lora and ControlNet, it takes a lot of time to generate image...
